Feat/#2200 allow modifying stake fraction

davecreaser opened the pull request:
## Description

- Create saga to allow modification of stake fraction.
- UI to follow in another PR.

Block ingestor PR here: https://github.com/JoinColony/block-ingestor/pull/223

Ticket here: https://github.com/JoinColony/colonyCDapp/issues/2200

## Testing

Create a fake UI to trigger the saga, and trigger it. The stake fraction for the staked expenditure extension should be updated in the db.

## Diffs

**New stuff** ✨

* New saga 'setStakeFraction'

Contributes to #2200

iamsamgibbs reviewed (changes requested):
Nice work on this Dave! All working as expected 👍 

<img width="319" alt="Screenshot 2024-05-17 at 16 12 33" src="https://github.com/JoinColony/colonyCDapp/assets/34915414/e82cc2ac-ca55-4b27-8f5c-6affcde2ffa0">

Just looks like a couple of unrelated commits got accidentally mixed into this, once those are resolved should be good to go!

iamsamgibbs commented:
Some code snippets to add to `TmpAdvancedPayments.tsx` for the sake of anyone else who wants to test:

`import { type SetStakeFractionPayload } from '~redux/sagas/expenditures/setStakeFraction.ts';`

```
  const setStakeFraction = useAsyncFunction({
    submit: ActionTypes.SET_STAKE_FRACTION,
    error: ActionTypes.SET_STAKE_FRACTION_ERROR,
    success: ActionTypes.SET_STAKE_FRACTION_SUCCESS,
  });

  const handleModifyStakeFraction = async () => {
    const payload: SetStakeFractionPayload = {
      colonyAddress: colony.colonyAddress,
      stakeFraction: '50',
    };

    await setStakeFraction(payload);
  };
  ```
  
  ` <Button onClick={handleModifyStakeFraction}>Modify stake fraction</Button>`

davecreaser commented:
> Nice work on this Dave! All working as expected 👍
> 
> <img alt="Screenshot 2024-05-17 at 16 12 33" width="319" src="https://private-user-images.githubusercontent.com/34915414/331647654-e82cc2ac-ca55-4b27-8f5c-6affcde2ffa0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5NjAyNzgsIm5iZiI6MTcxNTk1OTk3OCwicGF0aCI6Ii8zNDkxNTQxNC8zMzE2NDc2NTQtZTgyY2MyYWMtY2E1NS00YjI3LThmNWMtNmFmZmNkZTJmZmEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE3VDE1MzI1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmYThiZmQyZTVhMDY2YThjZmUyNWNhODM5YmZlMmNiY2UyOTRkMzBmNzM0ODJhOWY3NDgwYTIxMzg3N2YyN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nSOVb6UXOA541JVJHhhix_KFszqCH85m6N3-Fe_2L7U">
> Just looks like a couple of unrelated commits got accidentally mixed into this, once those are resolved should be good to go!

Cheers Sam! And thanks for catching those commits, I've rebased and removed them :)

rdig reviewed (approved):
Nicely done!

jakubcolony commented on src/i18n/en.json:
```diff
@@ -422,6 +424,10 @@
     "transaction.group.removeVerifiedMembers.description": "Remove verified members",
+    "transaction.group.setStakeFraction.title": "Set Stake Fraction",
+    "transaction.group.setStakeFraction.description": "Set Stake Fraction",
+    "transaction.ColonyClient.installExtension.title": "Install extension",
+    "transaction.ColonyClient.installExtension.description": "Install extension",
```
Nice catch!

jakubcolony reviewed (approved):
Changes look and work great, nice job @davecreaser 👍 

Considering it's resolving the first part of the issue, would you mind opening another one for the other subtask?

davecreaser commented:
> Changes look and work great, nice job @davecreaser 👍
> 
> Considering it's resolving the first part of the issue, would you mind opening another one for the other subtask?

@jakubcolony No problem. You mean opening another issue or another PR?
